03 - Ebu rebalancing (#6)
orbcollective/mpc-examples: pkg/mpc-examples/contracts/03-ebu-rebalancer/README.md

# EBU Rebalancer Controller

## Summary

> EBURebalancerController is a Managed Pool Controller that has the ability to pause swaps as well as promote rebalancing through the gradual lowering of swap fees. 

Reviewer: ```suggestion
EbuRebalancerController is a Managed Pool Controller that has the ability to pause swaps as well as promote rebalancing through the gradual lowering of swap fees. 
```

> EbuRebalancerController is a Managed Pool Controller that has the ability to pause swaps as well as promote rebalancing through the gradual lowering of swap fees. 

Reviewer: ```suggestion
EbuRebalancerController is a Managed Pool Controller designed to implement a pool that behaves as a static basket of tokens that can periodically rebalance its tokens back to its target weightings.
```

## Details

> The EBURebalancer is an extension of the NullController with added ability to rebalance the Managed Pool by enabling swaps, and updating swap fees gradually to a nominal value over the course of 7 days. This lowering of swap fees incentivises arbitrageurs to bring the pool back to its desired weight of 33/33/33.  

Reviewer: ```suggestion
The EbuRebalancer is a Managed Pool Controller that makes the pool a nominally static basket of tokens. It enables anyone to rebalance the Managed Pool by enabling swaps with an extremely high swap fee, ramping the fees down gradually to a nominal value over the course of 7 days. This lowering of swap fees incentivizes arbitrageurs to bring the pool back to its desired weight. 
```

Author: Why are we removing detail that the weights will be set to 33/33/33, since the user is not able to set custom weights

> The EbuRebalancer is a Managed Pool Controller that makes the pool a nominally static basket of tokens. It enables anyone to rebalance the Managed Pool by enabling swaps with an extremely high swap fee, ramping the fees down gradually to a nominal value over the course of 7 days. This lowering of swap fees incentivizes arbitrageurs to bring the pool back to its desired weight.

Reviewer: ```suggestion
The EbuRebalancer is a Managed Pool Controller that makes the pool a nominally static basket of tokens. It allows anyone to rebalance the Managed Pool by using the `rebalancePool` function. The rebalance works by enabling swaps with an extremely high swap fee and gradually ramping down the fees to a nominal (near-zero) value. The gradual swap fee decrease incentivizes arbitrageurs to bring the pool back to its desired weight by progressively creating minimally viable arbitrage opportunities.
```

## Access Control

> ### EBURebalancerController

Reviewer: ```suggestion
### EbuRebalancerController
```

> The controller has no access control and all callable functions can be executed by any account. There are constraints within the `rebalance()` function, as it can only be called once every `30 days`. The rebalance period is set for `7 days`. During this time period the ability to call `pausePool()` is locked. 

Reviewer: ```suggestion
The controller has no access control and all callable functions can be executed by any account. There are constraints within the `rebalancePool()` function, as it can only be called once every `30 days`. The rebalance period is set for `7 days`. During this time period the ability to call `pausePool()` is locked. 
```

> ### EBURebalancerFactory

Reviewer: ```suggestion
### EbuRebalancerFactory
```
